ccl/changefeedccl: TestChangefeedSchemaChangeAllowBackfill failed

ccl/changefeedccl.TestChangefeedSchemaChangeAllowBackfill [failed](https://tanzanite.cluster.engflow.com/invocations/default/dee15c41-e3e7-4d91-8989-c4f0f9d57321?testReportRun=22&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on release-24.3 @ [6e4fdd9720f4d0ccfe6c3a7c08120be446ccc736](https://github.com/cockroachdb/cockroach/commits/6e4fdd9720f4d0ccfe6c3a7c08120be446ccc736):


```
=== RUN   TestChangefeedSchemaChangeAllowBackfill
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedSchemaChangeAllowBackfill1932467273
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:2908: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedSchemaChangeAllowBackfill1932467273
--- FAIL: TestChangefeedSchemaChangeAllowBackfill (15.78s)
=== RUN   TestChangefeedSchemaChangeAllowBackfill/enterprise
    helpers_test.go:883: making server as secondary tenant
    helpers_test.go:962: making enterprise feed factory
    helpers_test.go:976: pgURL enterprise root
    --- FAIL: TestChangefeedSchemaChangeAllowBackfill/enterprise (15.74s)
=== RUN   TestChangefeedSchemaChangeAllowBackfill/enterprise/drop_column
    changefeed_test.go:2834: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:198
        	            				pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/changefeed_test.go:2834
        	Error:      	Received unexpected error:
        	            	expected
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBaseErr
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:242
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:201
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:266
        	            	  | github.com/cockroachdb/cockroach/pkg/util/timeutil.RunWithTimeout
        	            	  | 	pkg/util/timeutil/timeout.go:28
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:262
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:199
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloads
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:273
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedSchemaChangeAllowBackfill.func1.3
        	            	  | 	pkg/ccl/changefeedccl/changefeed_test.go:2834
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) expected
        	            	  |   drop_column: [1]->{"after": {"a": 1}, "updated": "1733924608033718827.0000000001"}
        	            	  |   drop_column: [2]->{"after": {"a": 2}, "updated": "1733924608033718827.0000000001"}
        	            	  | got
        	            	  |   drop_column: [1]->{"after": {"a": 1}, "updated": "1733924608033718827.0000000001"}
        	            	  |   drop_column: [3]->{"after": {"a": 3}, "updated": "1733924611928070727.0000000000"}
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestChangefeedSchemaChangeAllowBackfill/enterprise/drop_column
        --- FAIL: TestChangefeedSchemaChangeAllowBackfill/enterprise/drop_column (4.55s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedSchemaChangeAllowBackfill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45508

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -3036,7 +3036,6 @@ func TestChangefeedSchemaChangeAllowBackfill(t *testing.T) {
 				`drop_column: [2]->{"after": {"a": 2, "b": "2"}}`,
 			})
 			sqlDB.Exec(t, `ALTER TABLE drop_column DROP COLUMN b`)
-			sqlDB.Exec(t, `INSERT INTO drop_column VALUES (3)`)
 			ts := schematestutils.FetchDescVersionModificationTime(t, s.TestServer.Server, `d`, `public`, `drop_column`, 2)
 
 			// Backfill for DROP COLUMN b.
@@ -3047,7 +3046,8 @@ func TestChangefeedSchemaChangeAllowBackfill(t *testing.T) {
 					ts.AsOfSystemTime()),
 			})
 
-			// Insert 3 into drop_column
+			// Insert 3 into drop_column.
+			sqlDB.Exec(t, `INSERT INTO drop_column VALUES (3)`)
 			assertPayloadsStripTs(t, dropColumn, []string{
 				`drop_column: [3]->{"after": {"a": 3}}`,
 			})
@@ -3088,6 +3088,7 @@ func TestChangefeedSchemaChangeAllowBackfill(t *testing.T) {
 			// version 2. Then, when adding column c, it goes from 9->17, with the schema change being visible at
 			// the 7th step (version 15). Finally, when adding column d, it goes from 17->25 ith the schema change
 			// being visible at the 7th step (version 23).
+			// TODO(#142936): Investigate if this descriptor version hardcoding is sound.
 			dropTS := schematestutils.FetchDescVersionModificationTime(t, s.TestServer.Server, `d`, `public`, `multiple_alters`, 2)
 			addTS := schematestutils.FetchDescVersionModificationTime(t, s.TestServer.Server, `d`, `public`, `multiple_alters`, 15)
 			addTS2 := schematestutils.FetchDescVersionModificationTime(t, s.TestServer.Server, `d`, `public`, `multiple_alters`, 23)

